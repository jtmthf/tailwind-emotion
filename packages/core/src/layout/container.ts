import * as breakpoints from '../core/responsive';
import { ThemedUtil } from '../types';
import { apply } from '../utils';
import { mxAuto } from '../spacing';

export const container: ThemedUtil = theme => {
  const padding = theme.container?.padding;
  const defaultPadding =
    typeof padding === 'object' ? padding.default : padding;

  return apply(
    { width: '100%' },
    ...(['sm', 'md', 'lg', 'xl'] as const).map(screen =>
      breakpoints[screen]({
        maxWidth: theme.screens?.[screen],
        padding: typeof padding === 'object' ? padding[screen] : undefined,
      })(theme),
    ),
    theme.container?.center && mxAuto,
    { padding: defaultPadding },
  )(theme);
};
