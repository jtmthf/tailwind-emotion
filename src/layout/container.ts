import css from '@emotion/css';
import * as breakpoints from '../core/responsive';
import { Theme } from '../theme';

export const container = (theme: Theme) =>
  css(
    { width: '100%' },
    ...(['sm', 'md', 'lg', 'xl'] as const).map(screen =>
      breakpoints[screen](css({ maxWidth: theme.screens[screen] }))(theme),
    ),
  );
