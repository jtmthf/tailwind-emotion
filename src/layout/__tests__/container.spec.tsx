import { ThemeProvider } from 'emotion-theming';
import renderer from 'react-test-renderer';
import { container } from '../container';
import { defaultTheme } from '../../theme';

const Container = () => (
  <ThemeProvider theme={defaultTheme}>
    <div css={container} />
  </ThemeProvider>
);

test('container', () => {
  expect(renderer.create(<Container />).toJSON()).toMatchSnapshot();
});
