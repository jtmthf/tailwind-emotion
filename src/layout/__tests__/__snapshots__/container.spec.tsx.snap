// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container 1`] = `
.emotion-0 {
  width: 100%;
}

@media (min-width:640px) {
  .emotion-0 {
    max-width: 640px;
  }
}

@media (min-width:768px) {
  .emotion-0 {
    max-width: 768px;
  }
}

@media (min-width:1024px) {
  .emotion-0 {
    max-width: 1024px;
  }
}

@media (min-width:1280px) {
  .emotion-0 {
    max-width: 1280px;
  }
}

<div
  className="emotion-0"
/>
`;
